### PR TITLE
consensus: change logging and handling of height mismatch

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

So a closer inspection to this problem seems to reveal that we were producing error level logging for events that shouldn't be at error level. 

My approach to solving this at the moment is that only errors relevant to the higher level function should be propogated. As `handleMsg()` handles errors by passing them to the error level logger therefore the only errors that should be returned by the functions it calls should be those that warrant error-level logging. 

In this case the votes coming in for earlier heights are ignored and so just added to the debug level and not propogated all the way back to `handleMsg()`

Closes: #4936

### CODE REVIEW

**Comment by cmwaters** [comment]
I am going to first try rerun the testnet on top of this commit to see if this solves the problem before merging

File: consensus/state.go [hunk 1]
```diff
@@ -1815,15 +1812,15 @@ func (cs *State) addVote(
 
 	// A precommit for the previous height?
 	// These come in while we wait timeoutCommit
-	if vote.Height+1 == cs.Height {
-		if !(cs.Step == cstypes.RoundStepNewHeight && vote.Type == types.PrecommitType) {
-			// TODO: give the reason ..
-			// fmt.Errorf("tryAddVote: Wrong height, not a LastCommit straggler commit.")
-			return added, ErrVoteHeightMismatch
+	if vote.Height+1 == cs.Height && vote.Type == types.PrecommitType {
+		if !(cs.Step == cstypes.RoundStepNewHeight) {
```
Comment (erikgrinaker) [inline consensus/state.go:1816]: This can be changed to `!=` now.

File: consensus/state.go [hunk 2]
```diff
@@ -1815,15 +1812,15 @@ func (cs *State) addVote(
...
+			// Late precommit at prior height is ignored
+			cs.Logger.Debug("Precommit vote came in after commit timeout and has been ignored", "vote", vote)
```
Comment (erikgrinaker) [inline consensus/state.go:1818]: Maybe include vote height here.

File: consensus/state.go [hunk 3]
```diff
@@ -1813,9 +1813,10 @@ func (cs *State) addVote(
 	// A precommit for the previous height?
 	// These come in while we wait timeoutCommit
 	if vote.Height+1 == cs.Height && vote.Type == types.PrecommitType {
-		if !(cs.Step == cstypes.RoundStepNewHeight) {
+		if cs.Step != cstypes.RoundStepNewHeight {
 			// Late precommit at prior height is ignored
-			cs.Logger.Debug("Precommit vote came in after commit timeout and has been ignored", "vote", vote)
+			cs.Logger.Debug("Precommit vote came in after commit timeout and has been ignored",
+				"vote height", vote.Height, "vote", vote)
```
Comment (melekes) [inline consensus/state.go:1821]: vote already contains the height

plus, "vote height" will not work! because of the space ` `.

Comment (cmwaters) [reply consensus/state.go:1821]: okay, didn't know you couldn't do that

**Pull request merged**